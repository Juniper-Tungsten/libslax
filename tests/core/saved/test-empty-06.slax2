version 1.2;

ns my extension = "http://example.com";
ns exsl extension = "http://exslt.org/common";

main <out> {
    var $a = my:test();
    var $b = my:test(1, 2);
    
    <a> {
        copy-of $a;
    }
    <b> {
        copy-of $b;
    }
    <c> {
        copy-of my:one("test");
    }
    var $d := my:one("one", "two");
    <d> exsl:object-type($d);
}

function my:test ($min = 0, $max = 100) {
    result $max - $min;
}

function my:one ($name, $fish = "fish") {
    result {
        <name> $name;
        <fish> $fish;
    }
}
