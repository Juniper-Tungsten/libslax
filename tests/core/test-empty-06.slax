version 1.0;

ns my extension = "http://example.com";
ns exslt extension = "http://exslt.org/common";

match / {
    <out> {
	var $a = my:test();
	var $b = my:test(1, 2);
	<a> { copy-of $a; }
	<b> { copy-of $b; }

	<c> {
	    copy-of my:one("test");
	}
	var $d := my:one("one", "two");
	<d> {
	    expr exslt:object-type($d);
	}
    }
}

function my:test ($min = 0, $max = 100)
{
    result $max - $min;
}

function my:one ($name, $fish = "fish")
{
    result {
	<name> $name;
	<fish> $fish;
    }
}

