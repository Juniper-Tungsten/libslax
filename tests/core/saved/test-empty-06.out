<?xml version="1.0"?>
<out>
  <a>100</a>
  <b>1</b>
  <c>
    <name>test</name>
    <fish>fish</fish>
  </c>
  <d>node-set</d>
</out>
